### ISSUE TITLE
Moving a predicate after it has been deleted causes a panic

### ISSUE THREAD

**Issue by pawanrawal** [dgraph-io/dgraph#2112]
To reproduce:

1. Start multiple Dgraph server nodes. 
2. Load some data.
3. Delete a predicate. 
4. Try to move the predicate to another group using the `/moveTablet` API. 

The server which originally served the predicate ends up in a panic.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc0d897]

goroutine 878637 [running]:
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Item).Value(0x0, 0xc4200e6560, 0xc4200e6560, 0x1, 0x0, 0x1a099c0)
	/home/pawan/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/iterator.go:85 +0x37
github.com/dgraph-io/dgraph/worker.movePredicateHelper(0x1a13fc0, 0xc42ae018f0, 0xc481b3d2fc, 0x4, 0x2, 0x0, 0x0)
	/home/pawan/go/src/github.com/dgraph-io/dgraph/worker/predicate_move.go:146 +0x642
github.com/dgraph-io/dgraph/worker.(*grpcWorker).MovePredicate(0xc420378040, 0x1a13fc0, 0xc42ae018f0, 0xc46bb10ae0, 0xc420378040, 0xb7cf5d, 0x11c3001)
	/home/pawan/go/src/github.com/dgraph-io/dgraph/worker/predicate_move.go:291 +0x23b
github.com/dgraph-io/dgraph/protos/intern._Worker_MovePredicate_Handler(0x12b1fa0, 0xc420378040, 0x1a13fc0, 0xc42ae018f0, 0xc470832700, 0x0, 0x0, 0x0, 0x1d, 0x25)
	/home/pawan/go/src/github.com/dgraph-io/dgraph/protos/intern/internal.pb.go:2700 +0x276
google.golang.org/grpc.(*Server).processUnaryRPC(0xc420519c80, 0x1a174c0, 0xc4204c8d80, 0xc4bdb12640, 0xc4202140f0, 0x19fe288, 0x0, 0x0, 0x0)
	/home/pawan/go/src/google.golang.org/grpc/server.go:900 +0x9d3
google.golang.org/grpc.(*Server).handleStream(0xc420519c80, 0x1a174c0, 0xc4204c8d80, 0xc4bdb12640, 0x0)
	/home/pawan/go/src/google.golang.org/grpc/server.go:1122 +0x1528
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc42036c0f0, 0xc420519c80, 0x1a174c0, 0xc4204c8d80, 0xc4bdb12640)
	/home/pawan/go/src/google.golang.org/grpc/server.go:617 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/pawan/go/src/google.golang.org/grpc/server.go:615 +0xa1
```

We should not try to move a predicate that has been deleted but in any case we should never panic.

**Comment by pawanrawal** [comment]
This happens because we are trying to access `item.Value()` even if we get `ErrKeyNotFound` error from badger.

**Comment by pawanrawal** [comment]
Fixed in master with 77ad97efeea4ae0d4a15b3758e51cd2ee938c2da.
